newIDE/app/src/Hints/ObjectsAdditionalWork.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+import { t } from '@lingui/macro';
+import { type AlertMessageIdentifier } from '../MainFrame/Preferences/PreferencesContext';
+import newNameGenerator from '../Utils/NewNameGenerator';
+import { type MessageDescriptor } from '../Utils/i18n/MessageDescriptor.flow';
+
+/*
+ * Define additional logic which executes after an object/instance has been created.
+ * Also, InfoBar can be used which could notify users of the additional changes.
+ * Declare new identifier for infoBar in Mainframe/Preferences/PreferenceContext
+ * and add it in hints/explaination list.
+ */
+
+export type InfoBarDetails = {|
+  identifier: AlertMessageIdentifier,
+  message: MessageDescriptor,
+  touchScreenMessage: MessageDescriptor,
+|};
+
+type InfoBarEvent = 'onObjectAdded' | 'onInstanceAdded';
+
+export const onObjectAdded = (
+  object: gdObject,
+  layout: gdLayout,
+  project: gdProject
+): ?InfoBarDetails => {
+  const services = objectType[object.getType()];
+  if (services) {
+    services.onObjectAdded(object, layout, project);
+    return services.getInfoBarDetails('onObjectAdded');
+  }
+
+  return null;
+};
+
+export const onInstanceAdded = (
+  instance: gdInitialInstance,
+  layout: gdLayout,
+  project: gdProject
+): ?InfoBarDetails => {
+  const services =
+    objectType[layout.getObject(instance.getObjectName()).getType()];
```

Comment:
calling this "services" is a bit weird because it's called "objectType" as a variable. "Service" is also very generic, everything is a service in a way :) Can we think of a better name?
If your file is called "ObjectsAdditionalWork", then it must surely store an object called "objectsAdditionalWork", or something using the same vocabulary. Then if it's indexing some "additional work" by "object (type)", then when you're accessing it by giving a type, you're then getting something called... "additionalWork"?